android/app/src/main/java/com/mattermost/helpers/CustomPushNotificationHelper.java

```diff
@@ -227,6 +238,153 @@ public static void createNotificationChannels(Context context) {
         }
     }
 
+    public static boolean verifySignature(final Context context, String signature, String serverUrl, String ackId) {
+        if (signature == null) {
+            Log.i("Mattermost Notifications Signature verification", "No signature in the notification");
```

Comment:
Is this for backward compatibility with older servers? If so, can we add a comment and ticket to make signature verification mandatory after, say, next two ESRs?

Reply:
This is for compatibility with old push proxies. I can add a comment about that.